internal/services/pam/pam_test.go

```diff
@@ -72,66 +75,110 @@ func TestNewService(t *testing.T) {
 func TestAvailableBrokers(t *testing.T) {
 	t.Parallel()
 
-	client := newPamClient(t, nil)
+	tests := map[string]struct {
+		currentUserNotRoot bool
+
+		wantErr bool
+	}{
+		"Success getting available brokers": {},
+
+		"Error as only root can make that call": {currentUserNotRoot: true, wantErr: true},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
 
-	abResp, err := client.AvailableBrokers(context.Background(), &authd.Empty{})
-	require.NoError(t, err, "AvailableBrokers should not return an error, but did")
+			a := newAuthorizer(t, tc.currentUserNotRoot)
+			client := newPamClient(t, nil, &a)
```

Comment:
I tried this at first, but then I had to refactor because I needs to change the authorizer behaviour in the middle of the tests, when calling `authorizertests.SetCurrentRootAsRoot(&a, !tc.currentUserNotRoot)`.

Reply:
synced up on HO!